### PR TITLE
[Android] build shared libs

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. build shared libs (nnstreamer and subplugins)
2. clean up build script

TODO : we need to update mk files in android-res repo and sub-plugins

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
@anyj0527 Please verify SNPE. I updated other sub-plugins also.

File: api/android/api/src/main/jni/Android-snpe.mk [hunk 1]
```diff
@@ -40,9 +40,8 @@ include $(CLEAR_VARS)
...
 LOCAL_SRC_FILES := $(NNSTREAMER_FILTER_SNPE_SRCS)
-LOCAL_CXXFLAGS += -std=c++11 -frtti -fexceptions -Wno-exceptions -O2 -DNDEBUG $(NNS_API_FLAGS)
-LOCAL_C_INCLUDES := $(NNSTREAMER_INCLUDES) $(SNPE_INCLUDES) $(GST_HEADERS_COMMON)
-LOCAL_SHARED_LIBRARIES := libSNPE
-LOCAL_STATIC_LIBRARIES := nnstreamer
+LOCAL_CXXFLAGS := -std=c++11 -frtti -fexceptions -Wno-exceptions -O2 -DNDEBUG
+LOCAL_C_INCLUDES := $(SNPE_INCLUDES)
+LOCAL_SHARED_LIBRARIES := libSNPE $(NNS_API_LIBS)
 
 include $(BUILD_STATIC_LIBRARY)
```
Comment (anyj0527) [inline api/android/api/src/main/jni/Android-snpe.mk:47]: I verified with ` include $(BUILD_SHARED_LIBRARY) `.
You may change this line too!

File: api/android/api/src/main/jni/Android-snpe.mk [hunk 2]
```diff
@@ -40,9 +40,8 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE := snpe
 LOCAL_SRC_FILES := $(NNSTREAMER_FILTER_SNPE_SRCS)
-LOCAL_CXXFLAGS += -std=c++11 -frtti -fexceptions -Wno-exceptions -O2 -DNDEBUG $(NNS_API_FLAGS)
-LOCAL_C_INCLUDES := $(NNSTREAMER_INCLUDES) $(SNPE_INCLUDES) $(GST_HEADERS_COMMON)
-LOCAL_SHARED_LIBRARIES := libSNPE
-LOCAL_STATIC_LIBRARIES := nnstreamer
+LOCAL_CXXFLAGS := -std=c++11 -frtti -fexceptions -Wno-exceptions -O2 -DNDEBUG
```
Comment (wooksong) [inline api/android/api/src/main/jni/Android-snpe.mk:43]: Is -fPIC is not needded?

File: api/android/api/src/main/jni/Android-snpe.mk (hunk 1 above)
Comment (jaeyun-jung) [reply api/android/api/src/main/jni/Android-snpe.mk]: Updated. plz check it again.

File: api/android/api/src/main/jni/Android-snpe.mk (hunk 2 above)
Comment (jaeyun-jung) [reply api/android/api/src/main/jni/Android-snpe.mk]: Updated

File: api/android/api/src/main/jni/Android-snpe.mk (hunk 1 above)
Comment (anyj0527) [reply api/android/api/src/main/jni/Android-snpe.mk]: ✔ Checked  ✔

**Comment by jaeyun-jung** [comment]
PR updated (2020-05-15)
- add mk file for sub-plugin prebuilt libraries
- add missed header file and options to run api example

Also please see the patch in android-res repo (https://github.com/nnstreamer/nnstreamer-android-resource/pull/5)

**Pull request merged**